efficientdet/keras/efficientdet_arch_keras.py

```diff
@@ -210,3 +217,468 @@ def get_config(self):
+  box_class_repeats: number of "intermediate" layers
+  separable_conv: True to use separable_conv instead of conv2D
+  survival_prob: if a value is set then drop connect will be used with this probability
+  use_tpu: True to use TPU compatible functions
+
+  """
+  def __init__(self,
+               num_classes=80,
+               num_anchors=9,
+               num_filters=32,
+               min_level=3,
+               max_level=7,
+               is_training=False,
+               act_type='swish',
+               repeats=4,
+               separable_conv=True,
+               survival_prob=None,
+               use_tpu=False,
+               data_format='channels_last',
+               name='class_net', **kwargs):
+
+    super(ClassNet, self).__init__(name=name, **kwargs)
+
+    self.num_classes = num_classes
+    self.num_anchors = num_anchors
+    self.num_filters = num_filters
+    self.min_level = min_level
+    self.max_level = max_level
+    self.repeats = repeats
+    self.separable_conv = separable_conv
+    self.is_training = is_training
+    self.survival_prob = survival_prob
+    self.act_type = act_type
+    self.use_tpu = use_tpu
+    self.data_format = data_format
+    self.use_dc = survival_prob and is_training
+
+    self.conv_ops = []
+    self.bn_act_ops = []
+
+    for i in range(self.repeats):
+      # If using SeparableConv2D
+      if self.separable_conv:
+        self.conv_ops.append(tf.keras.layers.SeparableConv2D(
+            filters=self.num_filters,
+            depth_multiplier=1,
+            pointwise_initializer=tf.initializers.VarianceScaling(),
+            depthwise_initializer=tf.initializers.VarianceScaling(),
+            data_format=self.data_format,
+            kernel_size=3,
+            activation=None,
+            bias_initializer=tf.zeros_initializer(),
+            padding='same',
+            name=f'class-%d' % i))
+      # If using Conv2d
+      else:
+        self.conv_ops.append(tf.keras.layers.Conv2D(
+            filters=self.num_filters,
+            kernel_initializer=tf.random_normal_initializer(stddev=0.01),
+            data_format=self.data_format,
+            kernel_size=3,
+            activation=None,
+            bias_initializer=tf.zeros_initializer(),
+            padding='same',
+            name=f'class-%d' % i))
+
+      # Level only apply here so it's maybe better inside (no need to use tf.AUTO_REUSE anymore)
+      bn_act_ops_per_level = {}
+      for level in range(self.min_level, self.max_level + 1):
+        bn_act_ops_per_level[level] = keras.utils_keras.BatchNormAct(
+            self.is_training,
+            act_type=self.act_type,
+            init_zero=False,
+            use_tpu=self.use_tpu,
+            data_format=self.data_format,
+            name='class-%d-bn-%d' % (i, level))
+      self.bn_act_ops.append(bn_act_ops_per_level)
+
+    if self.use_dc:
+      self.dc = keras.utils_keras.DropConnect(self.survival_prob)
+
+    if self.separable_conv:
```

Comment:
Can you decide between regular and separable_conv once, instead of having 2 of these if blocks?

Reply:
I left it for retro-compatibility issues